packages/typespec-client-generator-core/src/http.ts

```diff
@@ -596,7 +596,11 @@ function getCollectionFormat(
         ? getHeaderFieldOptions(program, type)
         : undefined
   )?.format;
-  if (tspCollectionFormat === "form" || tspCollectionFormat === "simple") {
+  if (
+    tspCollectionFormat === "form" ||
+    tspCollectionFormat === "simple" ||
+    tspCollectionFormat === "csv"
```

Comment:
Why add filter logic for `csv` here?

Reply:
Because those are all the default and I tried to make it so there would not be a diff in the code model

Reply:
but yeah I think this was added first in order to not have `csv` show up all the time but changed it in the http library to only populate format: when the type was an array to more closely replicate when you would have used format.